src/libs/Middleware/SaveResponseInOnyx.js

```diff
@@ -13,23 +15,30 @@ function SaveResponseInOnyx(response, request) {
                 return;
             }
 
+            // For most requests we can immediately update Onyx. For write requests we queue the updates and apply them after the sequential queue has flushed to prevent a replay effect in
+            // the UI. See https://github.com/Expensify/App/issues/12775 for more info.
+            let updateHandler = Onyx.update;
+            if (request.data.apiRequestType === CONST.API_REQUEST_TYPE.WRITE) {
+                updateHandler = QueuedOnyxUpdates.queueOnyxUpdates;
+            }
+
             // First apply any onyx data updates that are being sent back from the API. We wait for this to complete and then
             // apply successData or failureData. This ensures that we do not update any pending, loading, or other UI states contained
             // in successData/failureData until after the component has received and API data.
             const onyxDataUpdatePromise = responseData.onyxData
-                ? Onyx.update(responseData.onyxData)
+                ? updateHandler(responseData.onyxData)
                 : Promise.resolve();
 
-            onyxDataUpdatePromise.then(() => {
+            return onyxDataUpdatePromise.then(() => {
                 // Handle the request's success/failure data (client-side data)
                 if (responseData.jsonCode === 200 && request.successData) {
-                    Onyx.update(request.successData);
-                } else if (responseData.jsonCode !== 200 && request.failureData) {
-                    Onyx.update(request.failureData);
+                    return updateHandler(request.successData);
                 }
-            });
-
-            return responseData;
+                if (responseData.jsonCode !== 200 && request.failureData) {
+                    return updateHandler(request.failureData);
+                }
+                return Promise.resolve();
```

Comment:
It it possible for us to ever get here?

Reply:
Yes, if none of the above conditions are met. If the `jsonCode` is non 200 and there's no `failureData` provided in the request.